tpr/tpr.go

```diff
@@ -0,0 +1,120 @@
+package tpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/rossf7/pingdom-operator/pkg/util"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	tprInitRetries    = 30
+	tprInitRetryDelay = 3 * time.Second
+)
+
+type Config struct {
+	// Dependencies.
+	Clientset kubernetes.Interface
+	Rest      rest.Interface
+
+	// Settings.
+	Name        string
+	Group       string
```

Comment:
Some comments around here would be useful. I am not sure what group is used for and which role name has. I can imagine that version also has to follow some specific scheme.